tests/json_test.cpp

```diff
@@ -40,11 +40,151 @@ TEST_CASE( "avoid_serializing_default_values", "[json]" )
 {
     std::ostringstream os;
     JsonOut jsout( os );
-    jsout.member( "foo", "foo", "foo" );
-    jsout.member( "bar", "foo", "bar" );
+    const std::string foo = "foo";
+    const std::string bar = "bar";
+    jsout.member( foo, foo, foo );
+    jsout.member( bar, foo, bar );
```

Comment:
I have no idea what was going wrong here.  I saw your Discord comments, and what you were observing made no sense.

Reply:
![image](https://user-images.githubusercontent.com/30208401/92862774-1dabdd00-f3c9-11ea-9576-532af8d98651.png)
for context for anyone else who might be able to chime in

Reply:
That's probably because the template function's deduced parameters are `const char *const &`, so the `==` operation compares the addresses instead of the values. (String literals of the same content are not required to have the same address in c++).

Reply:
Right.  That makes sense.  But they can optionally have the same address, which explains how it worked previously.